runtime/doc/lua.txt

```diff
@@ -1661,6 +1661,7 @@ select({items}, {opts}, {on_choice})                         *vim.ui.select()*
                                  • format_item (function item -> text)
                                    Function to format an individual item from
                                    `items` . Defaults to `tostring` .
+                                 • type (string|nil) Type of `items`.
```

Comment:
The docs here are generated from the lua docs in code. You'll have to update it in the ui.lua file

And I think this could also use some further explanation that clarifies how it's supposed to be used.